lib/Sema/CSSimplify.cpp

```diff
@@ -1232,7 +1233,7 @@ static bool isPotentiallyMoreOptionalThan(Type objType1,
 /// Enumerate all of the applicable optional conversion restrictions
 static void enumerateOptionalConversionRestrictions(
                     Type type1, Type type2,
-                    ConstraintKind kind,
+                    ConstraintKind kind, bool isFunctionInputOrResult,
```

Comment:
Instead of adding another flag would it make more sense if you look at the locator instead?